### PR TITLE
Fix nested elements in SVG

### LINKED ISSUE #56: Grouping elements in SVG
Trying to render the example from https://developer.mozilla.org/en-US/docs/Web/SVG/Element/g using;

```
svg
  [ viewBox "0 0 100 100"
  , createAttribute "xmlns" "http://www.w3.org/2000/svg"
  ]
  [ g
    [ fill "white", stroke "green", strokeWidth "5" ]
    [ circle' [ cx "40", cy "40", r "25" ]
    , circle' [ cx "60", cy "60", r "25" ]
    ]
  ]
```

Fails and gives back a blank SVG. The problem appears to be the nesting of the circles inside the `g` tag, without the grouping the circles render fine.

### PR DESCRIPTION
**Pull request by asvanberg**
Fixes #56

### CODE REVIEW

File: src/Flame/Html/Element.js [hunk 1]
```diff
@@ -167,9 +167,12 @@ exports.createSingleSvgNode = function (nodeData) {
 };
 
 function asSvg(elements) {
-    for (let e of elements)
+    for (let e of elements) {
         if (e.nodeType === elementNode)
             e.nodeType = svgNode;
+        if (typeof e.children !== 'undefined')
```
Comment (easafe) [inline src/Flame/Html/Element.js:173]: This won't protect against ```null```, will it? I think on JavaScript files we gotta check for both.

**Review by easafe — CHANGES_REQUESTED** [review]
Nice work @asvanberg . Could you add some tests as well?

File: src/Flame/Html/Element.js (hunk 1 above)
Comment (asvanberg) [reply src/Flame/Html/Element.js:173]: I got no errors on any of my SVG making use of all of `element`, `element'`, and `element_`. If you can tell me how make `children` null I can test it. `typeof null` in the console does give `'object'` so yeah it should fail if children can actually be null and not just undefined.

**Comment by asvanberg** [comment]
No idea how to write a test for this. Scanning over the both [Main.purs](https://github.com/easafe/purescript-flame/blob/master/test/Main.purs) and [Main.js](https://github.com/easafe/purescript-flame/blob/master/test/Main.js) gives nothing helpful about `nodeType` so nothing to learn from. The only tests related to SVG seems to be the rendered HTML (which was fine before this change already).

File: src/Flame/Html/Element.js (hunk 1 above)
Comment (easafe) [reply src/Flame/Html/Element.js:173]: Users of the library can't make ```children``` to be ```null```. Someone modifying the code in the future could :)

**Comment by easafe** [comment]
The tests are super messy, yeah. But for this kind of thing you can, say, mount the SVG snippet from the open issue and check that the elements were indeed created (e.g. https://github.com/easafe/purescript-flame/blob/master/test/Main.purs#L319 or https://github.com/easafe/purescript-flame/blob/master/test/Main.purs#L463)

**Pull request merged**